### PR TITLE
Make `ocf_version` const in OCF Manifest File

### LINKED ISSUE #382: [Enhancement]: Make Manifest Schema Version a Const
### Description of Enhancement :

The manifest version tag should be a constant.

### Why is this Needed?

Version certainty.

### Anything else we need to know?

No

### PR DESCRIPTION
**Pull request by pjohnmeyer**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

Prior to this change, `OCFVersionType` maintained a running list of current and past releases. This was misleading, because it:

a) implied a level of compatibility, both forward and backward, that
   is not guaranteed, and

b) implied that the value of this field did not need to match the
   version one is intending to communicate

This change eliminates OCFVersionType and sets `ocf_version` to a constant value in the OCF Manifest File. This also requires all sample files to be updated, which for now remains a manual step.

The document generator was also updated to allow `description` values on inline constants. Without this change the documentation for `ocf_version` just said "Scalar Constant" which was not helpful.

#### Which issue(s) this PR fixes:

Fixes #382

### CODE REVIEW

**Comment by pjohnmeyer** [comment]
**TWG**
- We will use `const` as originally proposed, not an enum
- Updating samples will be a manual step for now b/c there are only a few anyway
- Will write follow-up issue on validating samples across multiple versions of the schema:
  - do we fetch them from the remote url?
  - do we use git to fetch tags manually?
  - etc?

File: schema/enums/OCFVersionType.schema.json [hunk 1]
```diff
@@ -4,6 +4,16 @@
...
+  "oneOf": [
+    {
+      "description": "The current semantic versioning version (https://semver.org/spec/v2.0.0.html)",
+      "const": "1.0.1-alpha+main"
+    },
+    {
+      "description": "A placeholder version for documentation and tutorials, to simplify maintenance",
+      "const": "~~~ SAMPLE ~~~"
+    }
+  ],
```
[the remark this answers is not in the document]
Comment (pjohnmeyer) [reply schema/enums/OCFVersionType.schema.json]: Since we decided not to maintain this approach in the TWG meeting this comment is no longer meaningful.

**Comment by pjohnmeyer** [comment]
@JSv4 ready for review; once this is merged I believe we can do the next release.

**Comment by JSv4** [comment]
Will review.

**Review by jacobyavis — APPROVED** [review]
Looks good Patrick

**Pull request merged**